### PR TITLE
Users can see the health scores of every recipe

### PR DESCRIPTION
**Pull request by OluwadaraO**
Users can see the health scores on each recipe. These scores have different colors to show how good or bad it is for a user's health.
This is a user that wants a recipe high in protein. This recipe has a lot of fat hence the score of 50.4.
Screenshot 2024-07-16 at 9 12 50 PM

Also this user wanted food with low-carbs and this recipe contains high carbs
Screenshot 2024-07-16 at 9 13 06 PM

### CODE REVIEW

**Comment by joelmet-meta** [comment]
maybe you want to round to the nearest whole number for user experience?

File: culinary-canvas/capstone-project/backend/index.js [hunk 1]
```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
...
+    let lowestNutrient = null;
+    let lowestValue = Infinity;
+
+    Object.keys(nutrientTotals).forEach((key) => {
+        if(nutrientTotals[key] < lowestValue){
+            lowestNutrient = key;
+            lowestValue = nutrientTotals[key]
+        }
+    });
+    return lowestNutrient;
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js]: there should be a function for this. Maybe the reduce function here?: https://stackoverflow.com/questions/35527341/getting-all-minimal-values-from-dictionary-javascript

File: culinary-canvas/capstone-project/backend/index.js [hunk 2]
```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
...
+            subject: 'Try this recipe!',
+            message: `We noticed you needed more ${lowestNutrient}. So we're suggesting these recipes: `,
+            recipes: recipesHighInNutrient,
+            nutrientTotals : nutrientTotals
+        };
+    }catch(error){
+        console.error('Error fetching user notifications: ', error);
+        return{
+            subject: 'Error',
+            message: 'There was an error fetchig your notifications.',
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js]: *fetching

File: culinary-canvas/capstone-project/backend/index.js [hunk 3]
```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
...
+
+        const mealPlanner = await prisma.mealPlanner.findMany({
+            where: {userId : userId}
+        });
+
+        const recipeIds = mealPlanner.map(entry => entry.recipeId)
+
+        const fetchRecipeData = async(id)=>{
+            try{
+                const response = await fetch(`https://api.edamam.com/search?r=${encodeURIComponent(id)}&app_id=${EDAMAM_APP_ID}&app_key=${EDAMAM_APP_KEY}`);
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js:160]: is fetch a function you wrote somewhere? Or is this the whole API call, was just surprised there's not as much post-processing as I'd expect.

File: culinary-canvas/capstone-project/backend/index.js [hunk 4]
```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
...
+
+
+        const lowestNutrient = findLowestNutrient(nutrientTotals);
+        const recipesHighInNutrient = await fetchRecipesHighInNutrient(lowestNutrient)
+
+        return{
+            subject: 'Try this recipe!',
+            message: `We noticed you needed more ${lowestNutrient}. So we're suggesting these recipes: `,
+            recipes: recipesHighInNutrient,
+            nutrientTotals : nutrientTotals
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js:192]: This is a good idea!

File: culinary-canvas/capstone-project/backend/index.js [hunk 5]
```diff
@@ -57,6 +88,187 @@ const storage = new CloudinaryStorage({
...
+
+
+//to send the notifcation every 12am
+cron.schedule('0 0 * * *', async() => {
+
+    try{
+        const users = await prisma.users.findMany();
+        for (const user of users){
+            await sendNotification(user.id)
+        }
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js]: is there a way to map through the users asynchronously? Generally 'await' in a loop is bad practice.

File: culinary-canvas/capstone-project/backend/index.js [hunk 6]
```diff
@@ -252,17 +509,27 @@ app.get('/recipes', async (req, res) => {
...
+        const recipesWithHealthScores = data.hits.map(hit => {
+            const recipe = hit.recipe;
+            const healthInfo =calculateHealthScore(recipe, userPreferences);
+
+            return {
+                ...recipe,
+                likes:likesMap[hit.recipe.uri] || 0,
+                healthScore : healthInfo ? healthInfo.score : 0,
+                healthColor : healthInfo ? healthInfo.color : 0
+            }
```
Comment (joelmet-meta) [inline culinary-canvas/capstone-project/backend/index.js:523]: does this mean the same recipe could have different health scores for different users based on their diet and allergens?

**Review by joelmet-meta — APPROVED** [review]

File: culinary-canvas/capstone-project/backend/index.js (hunk 3 above)
Comment (OluwadaraO) [reply culinary-canvas/capstone-project/backend/index.js:160]: This is the API call

File: culinary-canvas/capstone-project/backend/index.js (hunk 4 above)
Comment (OluwadaraO) [reply culinary-canvas/capstone-project/backend/index.js:192]: Thank you!

File: culinary-canvas/capstone-project/backend/index.js (hunk 6 above)
Comment (OluwadaraO) [reply culinary-canvas/capstone-project/backend/index.js:523]: Yes, but all users that don't have allergens or diets would have recipes with the same health scores

**Comment by OluwadaraO** [comment]
The health scores are whole numbers: 
Screenshot 2024-07-17 at 12 01 42 PM

**Comment by OluwadaraO** [comment]
Users can see the bar chart which shows them their total Nutrients intake for the week(for meals they added to their weekly meal planner)
Screenshot 2024-07-17 at 8 03 52 PM
Screenshot 2024-07-17 at 12 07 19 PM

**Pull request merged**